File: src/main/java/com/will/utils/HttpPost.java
/**
 * 
 */
package com.will.utils;

import java.io.*;
import java.net.HttpURLConnection;
import java.net.URL;
import java.util.Map;

/**
 * HttpPost工具类
 * 
 * @author DongJun
 * 
 */
public class HttpPost {

	/**
	 * 以Post方式向web接口请求数据,编码为utf-8
	 * 
	 * 超时时间5s,编码UTF8
	 * 
	 * @param actionUrl
	 * @param params
	 * @return
	 * @throws IOException
	 * @throws Exception
	 */
	public static String post(String actionUrl, Map<String, Object> params)
			throws IOException {
		return post(actionUrl, params,5 * 1000);
	}
	
	/**
	 * 以Post方式向web接口请求数据,编码为utf-8
	 * 
	 * 编码UTF8
	 * 
	 * @param actionUrl
	 * @param params
	 * @param timeout 超时时间  毫秒
	 * @return
	 * @throws IOException
	 * @throws Exception
	 */
	public static String post(String actionUrl, Map<String, Object> params,int timeout)
			throws IOException {
		if (actionUrl == null || actionUrl.equals(""))
			throw new IllegalArgumentException("actionUrl");
		if (params == null)
			throw new IllegalArgumentException("params");
		String sb2 = "";
		String BOUNDARY = java.util.UUID.randomUUID().toString();
		String PREFIX = "--", LINEND = "\r\n";
		String MULTIPART_FROM_DATA = "multipart/form-data";
		String CHARSET = "UTF-8";
		// try {
		URL uri = new URL(actionUrl);
		HttpURLConnection conn = (HttpURLConnection) uri.openConnection();
		// 设置超时
		conn.setConnectTimeout(timeout);
		conn.setReadTimeout(timeout);
		conn.setDoInput(true);
		conn.setDoOutput(true);
		conn.setUseCaches(false);
		conn.setRequestMethod("POST");
		conn.setRequestProperty("connection", "keep-alive");
		conn.setRequestProperty("Charsert", "UTF-8");
		conn.setRequestProperty("Content-Type", MULTIPART_FROM_DATA
				+ ";boundary=" + BOUNDARY);

		StringBuilder sb = new StringBuilder();
		for (Map.Entry<String, Object> entry : params.entrySet()) {
			sb.append(PREFIX);
			sb.append(BOUNDARY);
			sb.append(LINEND);
			sb.append("Content-Disposition: form-data; name=\""
					+ entry.getKey() + "\"" + LINEND);
			sb.append("Content-Type: text/plain; charset=" + CHARSET + LINEND);
			sb.append("Content-Transfer-Encoding: 8bit" + LINEND);
			sb.append(LINEND);
			sb.append(entry.getValue());
			sb.append(LINEND);
		}

		DataOutputStream outStream = new DataOutputStream(
				conn.getOutputStream());
		outStream.write(sb.toString().getBytes("UTF-8"));

		byte[] end_data = (PREFIX + BOUNDARY + PREFIX + LINEND).getBytes();
		outStream.write(end_data);
		outStream.flush();

		InputStream in = null;
		int res = conn.getResponseCode();
		if (res == 200) {
			in = conn.getInputStream();
			BufferedReader br = new BufferedReader(new InputStreamReader(in,
					"UTF-8"));
			String line = "";
			for (line = br.readLine(); line != null; line = br.readLine()) {
				sb2 = sb2 + line;
			}
		}
		outStream.close();
		conn.disconnect();
		return sb2;
	}
}

File: src/main/java/com/will/utils/HttpPost.kt
package com.will.utils

import java.io.BufferedReader
import java.io.DataOutputStream
import java.io.InputStream
import java.io.InputStreamReader
import java.net.HttpURLConnection
import java.net.URL


/**
 * Created by zoumy on 2017/5/15 16:47.
 */
object HttpPost {

    /**
     * 以Post方式向web接口请求数据,编码为utf-8
     *
     * 超时时间5s,编码UTF8
     *
     * @param actionUrl
     * @param params
     * @return
     */

    fun post(actionUrl:String,param:Map<String,Any>,timeout:Int = 5000):String{
        if(actionUrl.isBlank()){
            throw IllegalArgumentException("action url : $actionUrl")
        }
        val BOUNDARY = java.util.UUID.randomUUID().toString()
        val PREFIX = "--"
        val LINEND = "\r\n"
        val MULTIPART_FROM_DATA = "multipart/form-data"
        val CHARSET = "UTF-8"

        val url = URL(actionUrl)

        val conn = url.openConnection() as HttpURLConnection
        conn.connectTimeout = timeout
        conn.readTimeout = timeout
        conn.doInput = true
        conn.doOutput = true
        conn.useCaches = false
        conn.requestMethod = "POST"
        conn.setRequestProperty("connection", "keep-alive")
        conn.setRequestProperty("Charsert", "UTF-8")
        conn.setRequestProperty("Content-Type", "$MULTIPART_FROM_DATA;boundary=$BOUNDARY")

        val sb = StringBuilder()
        param.forEach { t, u ->
            sb.append(PREFIX)
                    .append(BOUNDARY)
                    .append(LINEND)
                    .append("Content-Disposition: form-data; name=\"$t\"$LINEND")
                    .append("Content-Type: text/plain; charset=$CHARSET$LINEND")
                    .append("Content-Transfer-Encoding: 8bit$LINEND")
                    .append(LINEND)
                    .append(u)
                    .append(LINEND)
        }
        val outStream = DataOutputStream(conn.outputStream)
        outStream.write(sb.toString().toByteArray())

        val end_data = (PREFIX + BOUNDARY + PREFIX + LINEND).toByteArray()
        outStream.write(end_data)
        outStream.flush()

        val sb2 = StringBuilder()
        val input:InputStream
        val res = conn.responseCode
        if(res == 200){
            input = conn.inputStream
            val br = BufferedReader(InputStreamReader(input,"UTF-8"))
            br.forEachLine {
                sb2.append(it)
            }
        }

        outStream.close()
        conn.disconnect()
        return sb2.toString()

    }
}